hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/isauthorizedraw/IsAuthorizedRawCall.java

```diff
@@ -0,0 +1,203 @@
+        this.signature = requireNonNull(signature);
+
+        this.verificationStrategy = attempt.defaultVerificationStrategy();
+        this.sender = attempt.senderId();
+    }
+
+    @NonNull
+    @Override
+    public PricedResult execute(@NonNull final MessageFrame frame) {
+        requireNonNull(frame);
+
+        final var signatureType =
+                switch (signature.length) {
+                    case 65 -> SignatureType.EC;
+                    case 64 -> SignatureType.ED;
+                    default -> SignatureType.Invalid;
+                };
+
+        var accountNum = accountNumberForEvmReference(address, nativeOperations());
+        if (!isValidAccount(accountNum, signatureType)) {
+            return gasOnly(
+                    revertResult(INVALID_ACCOUNT_ID, gasCalculator.viewGasRequirement()), INVALID_ACCOUNT_ID, true);
+        }
+
+        boolean authorized = true;
+
+        // Validate parameters according to signature type
+        if (authorized) {
+            authorized = switch (signatureType) {
+                case EC -> messageHash.length == 32;
+                case ED -> true;
+                case Invalid -> false;};
+        }
+
+        // Gotta have an account that the given address is an alias for
+        final Optional<Account> account;
+        if (authorized) {
+            // TODO: Given code above account always exists
+            account = Optional.ofNullable(enhancement.nativeOperations().getAccount(accountNum));
+            authorized = account.isPresent();
+        } else account = Optional.empty();
+
+        // If ED then require a key on the account
+        final Optional<Key> key;
+        if (authorized && signatureType == SignatureType.ED) {
+            key = Optional.ofNullable(account.get().key());
+            authorized = key.isPresent();
+        } else key = Optional.empty();
+
+        // Key must be simple (for isAuthorizedRaw)
+        if (authorized && key.isPresent()) {
+            final Key ky = key.get();
+            final boolean keyIsSimple = !ky.hasKeyList() && !ky.hasThresholdKey();
+            authorized = keyIsSimple;
+        }
+
+        // Key must match signature type
+        if (authorized && key.isPresent()) {
+            authorized = switch (signatureType) {
+                case ED -> key.get().hasEd25519();
+                case EC -> false;
+                default -> false;};
+        }
+
+        if (authorized) {
+            authorized = switch (signatureType) {
+                case EC -> validateEcSignature(account.get());
+                case ED -> validateEdSignature(account.get(), key.get());
+                default -> false;};
+        }
+
+        final var gasRequirement = gasCalculator.gasCostInTinybars(HARDCODED_GAS_REQUIREMENT_GAS);
+
+        final var result = authorized
+                ? gasOnly(successResult(encodedAuthorizationOutput(authorized), gasRequirement), SUCCESS, false)
```

Comment:
we might want to set the last parameter - `isViewCall` to be true?  As this function does not change state.

Reply:
good idea